back/mysite3/views.py

```diff
@@ -1,45 +1,26 @@
 from django.db import connection
-from django.http import HttpResponse
 from django.views.generic import View
-from rest_framework import serializers
-from rest_framework.serializers import Serializer
 from polls.models import Question, Choice
-from rest_framework.decorators import api_view, permission_classes
+from rest_framework.decorators import api_view, permission_classes, action
 from rest_framework.permissions import AllowAny
 from rest_framework.response import Response
 from polls.serializers import ChoiceSerializer, QuestionSeralizer
 from rest_framework import viewsets
 from django.shortcuts import get_object_or_404
 
-
-@api_view(["get"])
-@permission_classes([AllowAny])
-def hello_world(request):
-    return Response({"message": "Hello, world!"})
-
-@api_view(["get"])
-@permission_classes([AllowAny])
-def q_list(request):
-    serializer = QuestionSeralizer(Question.objects.all())
-    return Response(serializer.data)
-
-@api_view(["get"])
-@permission_classes([AllowAny])
-def up_vote(request, choice_id):
-    choice = get_object_or_404(Choice, pk=choice_id)
-    choice.votes += 1
-    choice.save()
-    return Response("done")
-
-
 class QuestionViewSet(viewsets.ModelViewSet):
-    """
-    A simple ViewSet for listing or retrieving users.
-    """
+
     queryset = Question.objects.all()
     serializer_class = QuestionSeralizer
     permission_classes = [AllowAny]
 
+    @action(detail=False, methods=["get"])
+    def get_queryset(self, pk=None):
```

Comment:
So you don't want to do this because you're overriding the `get_queryset` method which is built in to django viewsets. In what situation do we only want 5 questions out of all of them? I think, in this case, we'll likely want to change how we're organizing the questions say relating them to a user?? Next stop will be to create users!! I don't think you want to have your list endpoint only list some subset of questions. This is also just a magic number on the backend. How do i know that i'm only getting the last 5 questions back??? We'd want an endpoint like `def list_n_questions(self, request):` which gets a body with the number of questions just like when users pass back the name of a question and the answers we can just have them pass the number of questions.